lib/main.dart

```diff
@@ -44,14 +45,25 @@ class MyApp extends StatelessWidget {
         ChangeNotifierProvider(create: (_) => UserModel.instance()),
         ChangeNotifierProvider(create: (_) {
           DateTime today = DateTime.now();
+          final firstDayOfTheWeek = today
+              .subtract(Duration(days: today.weekday))
+              .millisecondsSinceEpoch;
+          final lastDayOfTheWeek = today
+              .add(Duration(days: 7 - today.weekday))
+              .millisecondsSinceEpoch;
           DateTime firstDate =
               beginningOfDay(DateTime(today.year, today.month, 1));
+          Stream<List<Classes>> _classesStream =
+              ClassesRepository(FirebaseFirestore.instance)
```

Comment:
ClassRepository should be a global variable so you would only have to do `classRepo.getClassesOfTheDay(...)`.